circuits/cpp/src/aztec3/circuits/kernel/private/native_private_kernel_circuit.test.cpp

```diff
@@ -344,6 +345,277 @@ TEST_F(native_private_kernel_init_tests, private_function_incorrect_storage_cont
     EXPECT_EQ(composer.get_first_failure().message, "Storage contract address must be that of the called contract");
 }
 
+TEST_F(native_private_kernel_init_tests, native_read_request_bad_request)
+{
+    NT::fr const& amount = 5;
+    NT::fr const& asset_id = 1;
+    NT::fr const& memo = 999;
+
+    auto private_inputs = do_private_call_get_kernel_inputs_init(false, deposit, { amount, asset_id, memo });
+
+    auto const& contract_address =
+        private_inputs.private_call.call_stack_item.public_inputs.call_context.storage_contract_address;
+
+    auto [read_requests, read_request_membership_witnesses, root] = get_random_reads(contract_address, 2);
+    private_inputs.private_call.call_stack_item.public_inputs.historic_private_data_tree_root = root;
+
+    // tweak read_request so it gives wrong root when paired with its sibling path
+    read_requests[1] += 1;
+
+    private_inputs.private_call.call_stack_item.public_inputs.read_requests = read_requests;
+    private_inputs.private_call.read_request_membership_witnesses = read_request_membership_witnesses;
+
+    DummyComposer composer = DummyComposer("native_private_kernel_init_tests__native_read_request_bad_request");
+    auto const& public_inputs = native_private_kernel_circuit_initial(composer, private_inputs);
+
+    validate_no_new_deployed_contract(public_inputs);
+
+    ASSERT(composer.failed());
+    ASSERT_EQ(composer.get_first_failure().code,
+              CircuitErrorCode::PRIVATE_KERNEL__READ_REQUEST_PRIVATE_DATA_ROOT_MISMATCH);
+
+    // Check the first nullifier is hash of the signed tx request
+    ASSERT_EQ(public_inputs.end.new_nullifiers[0], private_inputs.signed_tx_request.hash());
+}
+
+TEST_F(native_private_kernel_init_tests, native_read_request_bad_leaf_index)
+{
+    NT::fr const& amount = 5;
```

Comment:
this setup logic is seen repeated alot, we should probably make these args the default in `do_private_call_get_kernel_inputs_init`